### PR TITLE
Simple header/footer

### PR DESCRIPTION
**Pull request by nicollecastrog**
The documentation I've added to the README explains my approach pretty thoroughly, but in essence I didn't want to make any presumptions about what the Header or Footer components would look like. All I wanted to solve was whether they were "fixed" or not, depending on the previous route's header/footer and the current route's header/footer. You'll notice that I created a new example application called "App", as I was having a hard time getting the older dependencies of the other example apps working with my version of Expo.

______________________

If the Header/Footer is the same between the two routes, then it will *not* be included in the transition animation (note how the header and footer do not animate when I tap on the "wallet" link below):

______________________

If a given route has a different Header/Footer, then said different element will be included in transition animation (I admit this looks weird below because I'm not doing any nice fading in/out upon mount/unmount of the tab navigation footer itself, but it's just for illustrative purposes):

It's worth noting this is a pretty common pattern. For example it's what WhatsApp does when transitioning between List view and Convo view: as each view has a different footer, the footer is included in the transition. The below screenshot was taken mid-swipe-back:

______________________

Finally, if a given route has *no* Header/Footer, then of course none is rendered for that route and the previous route's Header/Footer will be included inside *its* transition area (meaning they won't suddenly appear/disappear):

### CODE REVIEW

File: examples/App/.babelrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "presets": ["babel-preset-expo"],
+  "env": {
+    "development": {
+      "plugins": ["transform-react-jsx-source",
+        ["module-resolver", {
+          "alias": {
+            "react-router-native-stack": "../.."
```
Comment (nicollecastrog) [inline examples/App/.babelrc]: I followed steps [here](https://blog.callstack.io/adding-an-example-app-to-your-react-native-library-d23b9741a19c) to resolve "multiple require"-style errors and make this example `App` use imports the same way a "real" app using `react-router-native-stack` would do

File: examples/App/App.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+export default class App extends Component {
+  render() {
+    return (
+      <NativeRouter>
+        <Stack>
+          <Route exact path="/" component={Home} headerComponent={Header} />
+          <Route exact path="/fullscreen" component={FullScreen} />
+          <Route exact path="/page/pizza" component={Page} headerComponent={PizzaHeader} />
+          <Route path="/page/:name" component={Page} headerComponent={Header} />
```
Comment (nicollecastrog) [inline examples/App/App.js]: the usage of the `headerComponent` prop and the `Header` and `PizzaHeader` (along with some styling) are basically the only new bits here, everything else I copied from existing `Usage` example app

**Comment by nicollecastrog** [comment]
Once https://github.com/Traviskn/react-router-native-stack/pull/34 is merged, then the `yarn.lock`s won't show up in the diff here

File: examples/App/.babelrc (hunk 1 above)
Comment (Traviskn) [reply examples/App/.babelrc]: Nice!

**Comment by Traviskn** [comment]
Thanks for putting this together! I have a few thoughts on the API for adding a header/footer to the Stack. 

Firstly, if you need a "fixed" header/footer you can actually already do that by rendering it outside of the stack, either above or below it. 

So the only benefit you could get by having the header or footer rendered inside the stack is access to the same animated value that is driving the Stack's animation. If the header or footer could receive the internal `animatedValue` from the `StackTransitioner` component as a prop, then they could build any sort of animation they need an keep it in sync with the Stack's animated transitions. Would you mind passing down the `animatedValue` as a prop to the header/footer components as part of this patch?

Someday I think it would be nice to provide some standard header animations out-of-the-box similar to react-navigation, but this first step of just passing in a custom component is great.

**Comment by nicollecastrog** [comment]
Hi @Traviskn thanks for your comments! Glad you like the general idea of this PR 😄 

`Firstly, if you need a "fixed" header/footer you can actually already do that by rendering it outside of the stack, either above or below it.` 

^That's what I did initially, but then the header/footer will be present for *every* route. It doesn't take into account routes where I *don't* want said header/footer. So what I tried next was to continue to render the header/footer outside the stack and hook into the history via `withRouter` and then use some conditional logic to determine whether or not the component (the header or footer) should render itself, based on the current path. That works well enough if you don't have route transition animations. But the minute you're transitioning from a route *with* a header/footer to one *without* then AND there's a route transition animation in place, it makes the animation "jump" because the following happens:

* new child starts sliding in from the right (the animation area is still constrained within previous route's header/footer)
* header/footer suddenly disappear
* new child continues to slide in the from the right, but it "jumps" as the Stack suddenly has more vertical space to occupy

So I guess maybe I shouldn't have used the word "fixed" really! 😂 Because it's more like I'm trying to cater for situations where the header/footer is fixed between *certain routes* but not *all* of them.

Anyway, that's ^ why I couldn't render the header/footer outside the stack in my situation. And what inspired this PR. Maybe I shouldn't call it a "fixed" header/footer, because you're totally right in that a truly fixed (for the whole app) header/footer can be easily rendered outside the stack.

___________________________

LOVE ❤️ the idea of passing down the internal `animatedValue` to the header/footer components so that you can build up your own animations in sync with the transition animation itself. I'll add that in now 😄

**Comment by nicollecastrog** [comment]
@Traviskn done!

**Comment by Traviskn** [comment]
Thanks! Sorry for my delayed response

I think this is a great step towards supporting more dynamic headers/footers, thanks for all the work you've put in! 👍

**Pull request merged**